### PR TITLE
adding refactor code and shape

### PR DESCRIPTION
**Pull request by sperBruno**

### CODE REVIEW

**Comment by sperBruno** [comment]
Hi @carledriss could you take a look on this?

File: src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    public String getName() {
+        return customerName;
+    }
+
+    /**
+     * This method creates the ticket.
+     *
+     * @return the ticket
+     */
+    public String statement() {
```
Comment (carledriss) [inline src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java]: this method is doing 3 things. it is calculating the totalAmount, totalFrequentRenterPoints and printing.
Please split it.

File: src/main/java/com/jalasoft/selenium/bruno/movies/Childrens.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package com.jalasoft.selenium.bruno.movies;
+
+/**
+ * Created by Bruno Barrios on 3/6/2017.
+ */
+public class Childrens  extends Movie {
+
+    public static final int CHILDRENS_CODE = 2;
```
Comment (carledriss) [inline src/main/java/com/jalasoft/selenium/bruno/movies/Childrens.java]: Do we still need to have children code?

Comment (carledriss) [reply src/main/java/com/jalasoft/selenium/bruno/movies/Childrens.java]: I think we can remove it.

**Comment by sperBruno** [comment]
@carledriss, I have modified the code according your observations.

File: src/main/java/com/jalasoft/selenium/bruno/movies/Movie.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package com.jalasoft.selenium.bruno.movies;
+
+/**
+ * This is an abstract class for movies.
+ */
+public abstract class Movie {
+
+    private String titleMovie;
+    private int priceCode = 2;
```
Comment (carledriss) [inline src/main/java/com/jalasoft/selenium/bruno/movies/Movie.java]: priceCode will not be necessary if you are using inheritance

**Comment by sperBruno** [comment]
@carledriss, fixed with your observations. please take a look.

File: src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+        StringBuilder result = new StringBuilder();
+        result.append("Rental Record for " + getName() + BRAKE_LINE);
+        for (Rental rentals : rentalList) {
+            result.append(TABULATION + rentals.getMovie().getTitle() + TABULATION);
+        }
+        result.append("Amount owed is " + String.valueOf(calculateTotalAmount())).append(BRAKE_LINE);
+        result.append(BRAKE_LINE);
+        result.append("You earned " + String.valueOf(calculateTotalFrequentRentalPoints())).append(BRAKE_LINE);
```
Comment (carledriss) [inline src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java]: String.valueOf is not necessary in append method

File: src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+     * @return the ticket
+     */
+    public String statement() {
+
+        StringBuilder result = new StringBuilder();
+        result.append("Rental Record for " + getName() + BRAKE_LINE);
+        for (Rental rentals : rentalList) {
+            result.append(TABULATION + rentals.getMovie().getTitle() + TABULATION);
+        }
+        result.append("Amount owed is " + String.valueOf(calculateTotalAmount())).append(BRAKE_LINE);
```
Comment (carledriss) [inline src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java]: String.valueOf is not necessary in append method

File: src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+     * This method get the frequent renter points.
+     *
+     * @param frequentRenterPoints of a customer.
+     * @param rentals              made by customer.
+     * @return the frequent renter points.
+     */
+    public int getFrequentRenterPoints(final int frequentRenterPoints, final Rental rentals) {
+        int count = frequentRenterPoints;
+        count++;
+        if ((rentals.getMovie().getPriceCode() == 1)
```
Comment (carledriss) [inline src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java]: Why do we still need the getPriceCode method?
you need to take advantage from inheritance implementation here

File: src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+     * @return Total amount.
+     */
+    private double calculateTotalAmount() {
+        double totalAmount = 0;
+        double thisAmount;
+        for (Rental rentals : rentalList) {
+            thisAmount = rentals.getMovie().calculateMovieAmount(rentals.getDaysRented());
+            totalAmount += thisAmount;
+        }
+        return totalAmount;
```
Comment (carledriss) [inline src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java:99]: Try to use a stream here.
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html

File: src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+     * This method is going to calculate frequent rental points.
+     *
+     * @return total frequent rental points.
+     */
+    private double calculateTotalFrequentRentalPoints() {
+        int frequentRenterPoints = 0;
+        for (Rental rentals : rentalList) {
+            frequentRenterPoints = getFrequentRenterPoints(frequentRenterPoints, rentals);
+        }
+        return frequentRenterPoints;
```
Comment (carledriss) [inline src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java:112]: Try to use a stream here.
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html

File: src/main/java/com/jalasoft/selenium/bruno/movies/Movie.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
+package com.jalasoft.selenium.bruno.movies;
+
+/**
+ * This is an abstract class for movies.
+ */
+public abstract class Movie {
+
+    private String titleMovie;
+    private int priceCode;
```
Comment (carledriss) [inline src/main/java/com/jalasoft/selenium/bruno/movies/Movie.java]: We don't need the priceCode anymore
Please remove it.

File: src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+
+/**
+ * This class represents a Customer.
+ */
+class Customer {
+    private static final String BRAKE_LINE = "\n";
+    private static final String TABULATION = "\t";
+    private String customerName;
+    private List<Rental> rentalList = new ArrayList<>();
```
Comment (carledriss) [inline src/main/java/com/jalasoft/selenium/bruno/movies/Customer.java:14]: you should instantiate ArrayList in the constructor

**Pull request merged**